beluga_tutorial/README.md

```diff
@@ -0,0 +1,505 @@
+
+### 3.6 Measurement Update - Sensor Model
+Measurement models describe the formation process by which sensor measurements are generated in the physical world. Probabilistic robotics explicitly models the noise in sensor measurements. Such models account for the inherent uncertainty in the robot’s sensors. Formally, the measurement model is defined as a conditional probability distribution $p(z_t | x_t , m)$, where $x_t$ is the robot pose, $z_t$ is the measurement at time $t$, and $m$ is the map of the environment.
+
+The most common model for processing landmarks assumes that the sensor can measure the range and the bearing of the landmark relative to the robot’s
+local coordinate frame. In this tutorial we will use an approximation to the **landmark sensor model with known correspondence** (pag. 149 of ProbRob),
+since the doors do not have a signature that differentiates them from each other, but we can calculate the relative pose of each door to the robot. This sensor model, receives as inputs the landmark map, the sensor data and a particle (a state), and outputs the likelihood (or weight) of the sensor data assuming that the particle represent the true state.
+
+In our case, we need to generate simulated sensor data. This data corresponds to the relative positions of landmarks around the robot's **current_pose** within a configurable range specified by the **sensor_range** parameter.
+
+```cpp
+// Generate simulated sensor data
+auto sensor_data = landmark_map | ranges::views::remove_if([&](const double lm) {
+                      return std::abs(lm - current_pose) > tutorial_params.sensor_range;
+                    }) |
+                    ranges::views::transform([&](const double lm) { return lm - current_pose; }) |
+                    ranges::to<SensorData>;
+```
+* `landmark_map` is the map generated above.
+* `ranges::views::remove_if` will remove all landmarks grater than the range of view of the sensor.
+* `ranges::views::transform` transform them to relative poses from the robot's **current_pose**.
+* `ranges::to<SensorData>` convert the range to a SensorData type defined previously.
+
+To calculate the weight for each particle, we previously need to build the range `particle_sensor_data`. This range represent the relative distances of each landmark to the particle.
+
+```cpp
+// Sensor model
+auto sensor_model = [&](const double& state) {
+  auto particle_sensor_data =
+      landmark_map | ranges::views::transform([&](const double lm) { return lm - state; }) | ranges::to<SensorData>;
+
+  // Rest of the code
+  // ...
+};
+```
+The number of features identified at each time step is variable. However, many probabilistic robotic algorithms assume conditional independence between features, it means, the noise in each individual measurement is independent of the noise in other measurements. Under the conditional independence assumption, we can process one feature at-a-time. That is:
```

Comment:
> the noise in each individual measurement is independent of the noise in other measurements

@Alondruck ...when conditioned on the current system state and prior world knowledge.

Reply:
If you can check that part again to see if it is ok.